(class (attrs public ) org/python/modules/sets/PySetIterator
 (super org/python/core/PyObject)
 (source "PySetIterator.java")

 (field (attrs private ) _count int )
 (field (attrs private ) _iterator [object java/util/Iterator] )
 (field (attrs private ) _set [object java/util/Set] )

(method (attrs public ) <init>([object java/util/Set] )void
  (limit registers 3)
  ; this: v1 ([object org/python/modules/sets/PySetIterator])
  ; parameter[0] : v2 ([object java/util/Set])
  (line 17)
  (invoke-direct {v1} org/python/core/PyObject/<init> )
  (line 18)
  (iput-object v2 v1 org/python/modules/sets/PySetIterator._set [object java/util/Set])
  (line 19)
  (const/4 v0 0)
  (iput v0 v1 org/python/modules/sets/PySetIterator._count int)
  (line 20)
  (invoke-interface {v2} java/util/Set/iterator )
  (move-result-object v0)
  (iput-object v0 v1 org/python/modules/sets/PySetIterator._iterator [object java/util/Iterator])
  (line 21)
  (return-void )
 )

(method (attrs public ) __iter__()[object org/python/core/PyObject]
  (limit registers 1)
  ; this: v0 ([object org/python/modules/sets/PySetIterator])
  (line 24)
  (return-object v0)
)

(method (attrs public ) __iternext__()[object org/python/core/PyObject]
  (limit registers 3)
  ; this: v2 ([object org/python/modules/sets/PySetIterator])
  (catch java/util/ConcurrentModificationException from lb5274 to lb5286 using lb528a)
  (line 59)
  (iget-object v1 v2 org/python/modules/sets/PySetIterator._iterator [object java/util/Iterator])
  (invoke-interface {v1} java/util/Iterator/hasNext )
  (move-result v1)
  (if-eqz v1 lb529a)
  (line 60)
  (iget v1 v2 org/python/modules/sets/PySetIterator._count int)
  (add-int/lit8 v1 v1 1)
  (iput v1 v2 org/python/modules/sets/PySetIterator._count int)
  (label lb5274)
  (line 62)
  (iget-object v1 v2 org/python/modules/sets/PySetIterator._iterator [object java/util/Iterator])
  (invoke-interface {v1} java/util/Iterator/next )
  (move-result-object v1)
  (invoke-static {v1} org/python/core/Py/java2py [object java/lang/Object] )
  (label lb5286)
  (move-result-object v1)
  (label lb5288)
  (line 67)
  (return-object v1)
  (label lb528a)
  (line 63)
  (move-exception v0)
  (label lb528c)
  (line 64)
  (const-string v1 "dictionary changed size during iteration")
  (invoke-static {v1} org/python/core/Py/RuntimeError [object java/lang/String] )
  (move-result-object v1)
  (throw v1)
  (label lb529a)
  (line 67)
  (const/4 v1 0)
  (goto lb5288)
)

(method (attrs public ) next()[object org/python/core/PyObject]
  (limit registers 4)
  ; this: v3 ([object org/python/modules/sets/PySetIterator])
  (line 40)
  (invoke-virtual {v3} org/python/modules/sets/PySetIterator/__iternext__ )
  (move-result-object v0)
  (line 41)
  (if-nez v0 lb52fc)
  (line 42)
  (iget v1 v3 org/python/modules/sets/PySetIterator._count int)
  (iget-object v2 v3 org/python/modules/sets/PySetIterator._set [object java/util/Set])
  (invoke-interface {v2} java/util/Set/size )
  (move-result v2)
  (if-eq v1 v2 lb52ee)
  (line 45)
  (const-string v1 "dictionary changed size during iteration")
  (invoke-static {v1} org/python/core/Py/RuntimeError [object java/lang/String] )
  (move-result-object v1)
  (throw v1)
  (label lb52ee)
  (line 47)
  (const-string v1 "")
  (invoke-static {v1} org/python/core/Py/StopIteration [object java/lang/String] )
  (move-result-object v1)
  (throw v1)
  (label lb52fc)
  (line 49)
  (return-object v0)
)

)
